cloud-sql/postgres/Startup.cs

```diff
@@ -120,11 +102,57 @@ DbConnection NewPostgreSqlConnection()
             // closing idle connections in the pool if the count of all
             // connections exceeds MinPoolSize.
             connectionString.ConnectionIdleLifetime = 300;
-            // [END cloud_sql_postgres_dotnet_ado_lifetime]
+            // [END cloud_sql_postgres_dotnet_ado_lifetime]
+        }
+
+        DbConnection NewPostgreSqlTCPConnection()
+        {
+            // [START cloud_sql_postgres_dotnet_ado_connection_tcp]
+            var connectionString = new NpgsqlConnectionStringBuilder()
+            {
+                // Connecting to a local proxy that does not support ssl.
+                SslMode = SslMode.Disable,
+
+                Host = Environment.GetEnvironmentVariable("DB_HOST"),
+                Username = Environment.GetEnvironmentVariable("DB_USER"),
+                Password = Environment.GetEnvironmentVariable("DB_PASS"),
+                Database = Environment.GetEnvironmentVariable("DB_NAME"),
+            };
+            connectionString.Pooling = true;
+            // Specify additional properties here.
+            // [START_EXCLUDE]
+            SetDbConfigOptions(connectionString);
             // [END_EXCLUDE]
             NpgsqlConnection connection =
                 new NpgsqlConnection(connectionString.ConnectionString);
-            // [END cloud_sql_postgres_dotnet_ado_connection]
+            // [END cloud_sql_postgres_dotnet_ado_connection_tcp]
+            return connection;
+        }
+
+        DbConnection NewPostgreSqlUnixSocketConnection()
+        {
+            // [START cloud_sql_postgres_dotnet_ado_connection_socket]
+            String dbSocketDir = Environment.GetEnvironmentVariable("DB_SOCKET_PATH") ?? "/cloudsql";
+            String instanceConnectionName = Environment.GetEnvironmentVariable("INSTANCE_CONNECTION_NAME");
+            var connectionString = new NpgsqlConnectionStringBuilder()
+            {
+                // Connecting to a local proxy that does not support ssl.

```

Comment:
nit: (in a couple other places as well) 
```suggestion
                // The Cloud SQL proxy provides encryption between the proxy and service. 

```

Reply:
fixed